keanu-project/src/main/java/io/improbable/keanu/tensor/dbl/DoubleTensor.java

```diff
@@ -92,6 +93,21 @@ static DoubleTensor scalar(double scalarValue) {
         return new ScalarDoubleTensor(scalarValue);
     }
 
+    /**
+     * @param toPile    an array of DoubleTensor
+     * @return  a DoubleTensor with toPile joined along a new dimension 0
+     */
+    static DoubleTensor pile(DoubleTensor... toPile) {
```

Comment:
is `pile` standard terminology? Is this what's called `stack` in TensorFlow?

Reply:
I am calling it `pile` because it's nd4j calls it that for when you are stacking everything at dimension 0

Reply:
renamed it to `stack` and to take a dimension parameter so it behaves like `numpy.stack`.